haskell/compile.bzl

```diff
@@ -547,13 +543,19 @@ def _compile_module(
     compile_args_for_file.add("-o", objects[0].as_output())
     compile_args_for_file.add("-ohi", his[0].as_output())
     compile_args_for_file.add("-stubdir", stubs.as_output())
+    compile_args_for_file.add(packages_info.nix_packages)
 
     if link_style in [LinkStyle("static_pic"), LinkStyle("static")]:
         compile_args_for_file.add("-dynamic-too")
         compile_args_for_file.add("-dyno", objects[1].as_output())
         compile_args_for_file.add("-dynohi", his[1].as_output())
 
     compile_args_for_file.add(module.source)
+
+    aux_deps = ctx.attrs.srcs_deps.get(module.source)
+    if aux_deps:
+        compile_args_for_file.hidden(aux_deps)
+
     for (path, src) in srcs_to_pairs(ctx.attrs.srcs):
```

Comment:
Would these `aux_deps` suffice to replace this `srcs_to_pairs` loop?

Reply:
Yes, that's what I thought too, but I was a bit reluctant to remove this just yet. Maybe we should warn about other sources other than haskell being listed here and advice on using `srcs_deps` instead?